e2e/libs/volume/crd.py

```diff
@@ -174,3 +177,40 @@ def check_data(self, volume_name, checksum):
             f"md5sum {endpoint} | awk \'{{print $1}}\'")
         if _checksum != checksum:
             Exception(f"data was changed: {_checksum}/{checksum}")
+
+    def wait_delete_pod(self,api, pod_uid, namespace='default'):
+        for i in range(DEFAULT_POD_TIMEOUT):
+            ret = api.list_namespaced_pod(namespace=namespace)
+            found = False
+            for item in ret.items:
+                if item.metadata.uid == pod_uid:
+                    found = True
+                    break
+            if not found:
+                break
+            time.sleep(DEFAULT_POD_INTERVAL)
+        assert not found
+
+    def delete_and_wait_pod(self, volume_name):
+        vol_client = get_longhorn_client()
+        vol = vol_client.by_id_volume(volume_name)
+        eim_name = vol.controllers[0].instanceManagerName
+
+        target_pod = None
+        api = k8s_core_api()
+        namespace = 'longhorn-system'
+        try:
+            target_pod = api.read_namespaced_pod(name=eim_name,
+                                                namespace=namespace)
+        except ApiException as e:
+            assert e.status == 404
```

Comment:
When will this happen? What cause this happen?

Reply:
Reused code from https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/common.py#L574 ,with the hope, it's well tested.